Sources/ComposableArchitecture/Effects/Throttling.swift

```diff
@@ -19,34 +19,41 @@ extension Effect {
     scheduler: S,
     latest: Bool
   ) -> Effect where S: Scheduler {
-    self.flatMap { value -> AnyPublisher<Output, Failure> in
-      guard let throttleTime = throttleTimes[id] as! S.SchedulerTimeType? else {
-        throttleTimes[id] = scheduler.now
-        throttleValues[id] = nil
-        return Just(value).setFailureType(to: Failure.self).eraseToAnyPublisher()
-      }
+    self.receive(on: scheduler)
```

Comment:
An alternative solution would be to add the `receive(on:)` to each of the first the two exit paths (`Just(value)` calls that return immediately). 

The current approach ensures any value returned from the `flatMap` is always delivered on `scheduler` (unless explicitly scheduled elsewhere). However, we incur in an extra dispatch (and minor performance penalty) on the exit path where the value is delayed, because it is then dispatched by `delay` once again. My reasoning was tothat it was worth it to keep things simple, but can definitely change it if you prefer.